src/shared_realm.hpp

```diff
@@ -158,10 +164,12 @@ class Realm : public std::enable_shared_from_this<Realm> {
         // Optional schema for the file.
         // If the schema and schema version are supplied, update_schema() is
         // called with the supplied schema, version and migration function when
-        // the Realm is actually opened and not just retreived from the cache
+        // the Realm is actually opened and not just retrieved from the cache
         util::Optional<Schema> schema;
         uint64_t schema_version = -1;
         MigrationFunction migration_function;
+        // Called when the schema changes are delivered to this Realm instance.
+        SchemaChangedFunction schema_changed_function;
```

Comment:
I think this would make more sense as a member function on `BindingContext`. Based on the doc comment for `update_schema_notify()` it would remove the need for the `notify` parameter, and currently things are split up such that user-provided callbacks are supplied on the `Config` object, while callbacks from the object store to binding are on the `BindingContext` (which is not a split I ever considered before now, but now that I've noticed it seems sensible?).